pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -161,7 +163,8 @@ func (s *SSOSettingsService) Upsert(ctx context.Context, settings models.SSOSett
 	}
 
 	go func() {
-		err = social.Reload(context.Background(), settings)
+		settings.Settings = overrideMaps(storedSettings.Settings, settings.Settings, secrets)
+		err = social.Reload(context.Background(), *settings)
```

Comment:
it is a bit unfortunate that `Reload` can fail, especially when running as a separate goroutine, this could lead to a bit unexpected behaviour.  Ideally if we pass validation and update to database this should be a infallible call

Reply:
Yep, it shouldn't fail by default for OAuth providers, but it could fail for SAML when the data is coming from the `system` (which is the `settingsprovider` service in case of SAML). 

We'll consider improving this!